types/tx.go

```diff
@@ -1,3 +1,24 @@
 package types
 
+import (
+	"github.com/tendermint/go-merkle"
+)
+
 type Tx []byte
+
+type Txs []Tx
+
+func (txs Txs) Hash() []byte {
```

Comment:
I don't think this does what we want it to do... case 1: txs[0] should be a hash of txs[0]... otherwise SimpleHashFromTwoHashes(left,right) is actually hashing two large objects in the leaf-1 nodes.  Also, can we change the impl of go-merkle SimpleHashFromBinaries // SimpleHashFromHashables instead?